models/pull.go

```diff
@@ -889,3 +889,11 @@ func (pr *PullRequest) GetWorkInProgressPrefix() string {
 	}
 	return ""
 }
+
+// GetPullRequestByHeadBranch returns a pr by head branch
+func GetPullRequestByHeadBranch(headBranch string, repo *Repository) (pr *PullRequest, err error) {
+	pr = &PullRequest{}
```

Comment:
```suggestion
	pr = &PullRequest{HasMerged: false}
```
Maybe also add the PR status as a condition? Ideally, joining with `issue` and check if the issue is open too.

Reply:
But the name of the function suggests it will only return a pull request by head branch, it does not tell it would only get the pr if it is already closed. IMHO it is the callers responsibility to check this. That way, it can be used for multiple purposes, not only mine.

I'll add a check in `services/automerge/pull_auto_merge.go:45`

Reply:
Done.